File: src/types/workout.ts
export interface Workout {
  id: number;
  date: string;
  type: string;
  duration: number;
  caloriesBurned: number;
}

export type WorkoutFormData = Omit<Workout, 'id'>;

export interface ApiError {
  message: string;
  statusCode: number;
}

File: backend/FitnessTracker.Api/Models/Workout.cs
namespace FitnessTracker.Api.Models;

public class Workout
{
    public int Id { get; set; }
    public DateTime Date { get; set; }
    public string Type { get; set; } = string.Empty;
    public int Duration { get; set; }
    public int CaloriesBurned { get; set; }
}